server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -158,6 +158,30 @@ public void registerShuffle(
     String remoteStoragePath = req.getRemoteStorage().getPath();
     String user = req.getUser();
 
+    if (req.getIsStageRetry()) {
```

Comment:
If `removeShuffleDataSync` is always being called, we can avoid adding plumbing `isStateRetry` in here. When `isStateRetry == false`, this is a NOOP.

Method `removeShuffleDataSync` might return `true` if it found data to delete, so we can conditionally log the message below.

Reply:
I prefer reserving the `isStageRetry`(or use stage attempt number to replace this) param for 2 reasons

1. this is more explicit for stage retry, especially when something go wrong, like the previous data has been purged due to expire heartbeat. If having this, the log will indicate the abnormal problem happens
2. for the next PR, I will introduce the stage latest attempt to discard the older attempt data.